atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/OracleConnectionConfig.java

```diff
@@ -72,12 +74,17 @@ public String getUrl() {
     @Value.Derived
     @JsonIgnore
     public Optional<String> namespace() {
-        if (getSid().isPresent()) {
-            return getSid();
-        }
-        return serviceNameConfiguration().map(ServiceNameConfiguration::namespaceOverride);
+        return Stream.of(
+                        namespaceOverride(),
+                        getSid(),
+                        serviceNameConfiguration().map(ServiceNameConfiguration::namespaceOverride))
+                .filter(Optional::isPresent)
```

Comment:
this is clever, but a bit arcane - worth a javadoc here explaining the order of precedence.

Reply:
yep, will add